OFEP-metric-hooks.md

```diff
@@ -0,0 +1,67 @@
+## OFEP-hooks-for-metrics
+
+## State: PENDING REVIEW
+
+This OFEP propose to introduce OpenFeature hook for OpenTelemetry metrics.
+
+## Background
+
+We already have OpenTelemetry Span support through hooks. Similarly, we can provide a dedicated hook for 
+OpenTelemetry 
+metrics. Providing telemetry data out of the box will make OpenFeature attractive for both users and vendors. 
+
+## Proposal
+
+The proposal here is to define a set of metrics that can be used with different hook stages. For example, error metrics 
+can be collected at the hook's `error` stage. Going with this background, I propose the following metrics.
+
+### evaluationRequests
+
+A counter[1] based metric to calculate the number of flag evaluation requests. This is recorded at `before` stage of 
+the hook.
+
+### evaluationSuccess
+
+A counter[1] based metric to calculate the number of successful flag evaluations. This is recorded at `after` stage of 
+the hook.
+
+### evaluationError
+
+A counter[1] based metric to calculate the number of failed flag evaluations. This is recorded at `error` stage of
+the hook.
+
+### evaluationActive
```

Comment:
Evaluation latency metric as well?

Reply:
I would imagine that evaluation latency should be extremely low, but I suppose it would depend on the provider. It certainly could be an interesting and useful metric. Perhaps the hook could allow you to opt-in or out of specific metrics.

Reply:
@justinabrahms  good point. I considered this option but I think this is not possible as hooks have a global presence compared to for example a HTTP request (request/response roundtrip calculation). Hooks have access to context details, but there is no way to uniquely separate an identifier between stages (for latency this is needed between the `before` & `after` stage). 

For example, if we consider the flag key, there could be parallel flag evaluations for the same flag. In Go, there is the `context` that can be used as a carrier for evaluation start time which is shared with all stages, but this is language specific and could be challenging to implement in other languages.

This is why I opted for `evaluationActive` which is simple and contains the number of active flag evaluations with dimensions for better slicing. 

Anyway, I am open to other opinion and let me know if you see any other way to approach this :)

Reply:
The sdk could add a time stamp on the evaluation context object and the hook could calculate a diff from that. I'm not sure if we would want to do that but it seems like an option.

Reply:
I think I will resolve this with this change [1] and we can revisit this once we have the first implemention :) 

[1] - https://github.com/open-feature/ofep/pull/62/files#diff-1ee0937abd11c52172d0be2b854ab5162b15262da53e7ddbcd4090484879bc67R66